### ISSUE TITLE
Gradient Checkpointing

### ISSUE THREAD

**Issue by maxwxzheng** [lukemelas/EfficientNet-PyTorch#284]
Does anyone have any thoughts on where and how to add gradient checkpoints in this implementation to reduce memory usage during training?
